plugins/default/lib/test_mongodb_health.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+import unittest
+
+from mongodb_health import analize, parse_mongo_result, report
+
+
+class TestAnalize(unittest.TestCase):
+
+    def runTest(self):
+        self.assertEqual(analize(1), 0)
+        self.assertEqual(analize(2), 0)
+        self.assertEqual(analize(0), 1)
+        self.assertEqual(analize(7), 1)
+        self.assertEqual(analize(3), 2)
+        self.assertEqual(analize(5), 2)
+        self.assertEqual(analize(7), 1)
+        self.assertEqual(analize(9), 2)
+        self.assertEqual(analize(None), 3)
+
+
+class TestParseMongoResult(unittest.TestCase):
+
+    def runTest(self):
+        self.assertEqual(parse_mongo_result("0\n"), 0)
+        self.assertEqual(parse_mongo_result("1\n"), 1)
+        self.assertEqual(parse_mongo_result("2\n"), 2)
+        self.assertEqual(parse_mongo_result("3\n"), 3)
+        self.assertEqual(parse_mongo_result("5\n"), 5)
+        self.assertEqual(parse_mongo_result("7\n"), 7)
+        self.assertEqual(parse_mongo_result("9\n"), 9)
+        self.assertEqual(parse_mongo_result("10\n"), 10)
+        self.assertEqual(parse_mongo_result("\n"), None)
+        self.assertEqual(parse_mongo_result(""), None)
+
+
+class TestReport(unittest.TestCase):
+
+    def runTest(self):
+        self.assertEqual(report(
+            ("mongodb-1-1-s1ngl",),
+            (1,),
+            (0,)), 0)
```

Comment:
The choice of indentation of these assertions makes it harder to understand what is the input to a function call, and what is the expected value.

Suggestion:

```python
        self.assertEqual(
            report(
                ("mongodb-1-1-s1ngl",),
                (1,),
                (0,),
            ), nagios.OK)
```

What does that `(1,)` and `(0,)` mean?

Maybe I'd even suggest using keyword arguments here to make it a bit clearer:

```python
        self.assertEqual(
            report(
                pods=("mongodb-1-1-s1ngl",),
                rs_statuses=(PRIMARY,),
                nagios_statuses=(nagios.OK,),
            ), nagios.OK)
```

And now, after the rewrite, it makes me think whether this report should really result in `nagios.OK`.
It means that we have a replica set with only one primary and no secondaries... for me the output here should be at least `nagios.WARN`.